CMakeLists.txt

```
@@ -238,9 +238,11 @@ endif()
 option(INSTALL_TESTS "Install tests" OFF)
 option(BUILD_LAYERS "Build layers" ON)
 option(BUILD_LAYER_SUPPORT_FILES "Generate layer files" OFF) # For generating files when not building layers
-option(USE_ROBIN_HOOD_HASHING "Use robin-hood-hashing" ON)
+
+find_package(robin_hood CONFIG)
+option(USE_ROBIN_HOOD_HASHING "robin_hood provides faster versions of std::unordered_map and std::unordered_set" ${robin_hood_FOUND})
 if (USE_ROBIN_HOOD_HASHING)
-    find_package(robin_hood REQUIRED CONFIG)
+    add_definitions(-DUSE_ROBIN_HOOD_HASHING)
```

`add_definitions` isn't recommended anymore with modern CMake. Use `target_compile_definitions`

```suggestion
    target_compile_definitions(robin_hood::robin_hood INTERFACE USE_ROBIN_HOOD_HASHING)
```

This will apply the macro to any libraries that link against `robin_hood::robin_hood`